django/cantusdb_project/main_app/views/source.py

```diff
@@ -191,35 +191,56 @@ def test_func(self):
             return False
 
     def get_success_url(self):
-        return reverse("source-create")
+        return reverse("source-detail", args=[self.object.id])
 
     def form_valid(self, form):
         form.instance.created_by = self.request.user
-        source = form.save()
+        self.object = form.save()
```

Comment:
this looks fine, but I'm just curious - why was it necessary to substitute `self.object` for `source` here?

Reply:
@jacobdgm See the definition of `form_valid` in django's ModelFormMixin
```    
def form_valid(self, form):
        """If the form is valid, save the associated model."""
        self.object = form.save()
        return super().form_valid(form)
```

since `form_valid` is overwritten in our code, self.object is never saved and so we cannot access self.object in `get_success_url`. To avoid passing `source` as an argument to `get_success_url` I just replicated the behaviour from the ModelFormMixin `form_valid`